src/containers/BuyManaWithFiatModal/BuyManaWithFiatFeedbackModal/BuyManaWithFiatFeedbackModal.tsx

```diff
@@ -1,34 +1,110 @@
 import React from 'react'
-import { Network } from '@dcl/schemas'
 import {
   FeedbackModal as BaseBuyManaWithFiatFeedbackModal,
-  FeedbackModalI18N
+  FeedbackModalI18N,
+  TransactionStatus
 } from 'decentraland-ui/dist/components/BuyManaWithFiatModal/FeedbackModal'
+import {
+  gatewaysNames,
+  networksNames
+} from 'decentraland-ui/dist/components/BuyManaWithFiatModal/Network'
+import { getAnalytics } from '../../../modules/analytics/utils'
+import { Purchase, PurchaseStatus } from '../../../modules/mana/types'
 import { t } from '../../../modules/translation/utils'
 import { Props } from './BuyManaWithFiatFeedbackModal.types'
 
+const transactionStatuses = {
+  [PurchaseStatus.PENDING]: TransactionStatus.PENDING,
+  [PurchaseStatus.COMPLETE]: TransactionStatus.SUCCESS,
+  [PurchaseStatus.FAILED]: TransactionStatus.FAILURE,
+  [PurchaseStatus.CANCELLED]: TransactionStatus.FAILURE
+}
+
+const camelToSnakeCase = (str: string) =>
+  str.replace(/[A-Z]/g, letter => `_${letter.toLowerCase()}`)
+
+const propsToTranslateByStatus = {
+  [TransactionStatus.PENDING]: [
+    'title',
+    'statusTitle',
+    'description',
+    'goToText'
+  ],
+  [TransactionStatus.SUCCESS]: ['title', 'description', 'cta'],
+  [TransactionStatus.FAILURE]: [
+    'title',
+    'statusTitle',
+    'description',
+    'cta',
+    'secondaryCta'
+  ]
+}
+
 const getDefaultFeedbackTranslations = (
-  network: Network
-): FeedbackModalI18N => ({
-  title: t(
-    `@dapps.buyManaWithFiat.feedback_modal.${network.toLowerCase()}.title`
-  ),
-  description: t(
-    `@dapps.buyManaWithFiat.feedback_modal.${network.toLowerCase()}.description`
-  ),
-  cta: t(`@dapps.buyManaWithFiat.feedback_modal.${network.toLowerCase()}.cta`),
-  error: ''
-})
+  { network, gateway }: Purchase,
+  status: TransactionStatus
+): FeedbackModalI18N => {
+  const basePath = `@dapps.buyManaWithFiat.feedback_modal.${status}`
+  return Object.fromEntries(
+    propsToTranslateByStatus[status].map(it => [
+      it,
+      t(`${basePath}.${camelToSnakeCase(it)}`, {
+        network: networksNames[network],
+        gateway: gatewaysNames[gateway]
+      })
+    ])
+  ) as FeedbackModalI18N
+}
 
 const BuyManaWithFiatFeedbackModal = ({
-  metadata: { purchase },
+  metadata: { purchase, goToUrl },
+  onTryAgain,
+  onSelectOtherProvider,
   onClose
 }: Props) => {
+  const { network, gateway, status: purchaseStatus } = purchase
+  const transactionStatus = transactionStatuses[purchaseStatus]
+  const analytics = getAnalytics()
+
+  const handleCtaClick = () => {
```

Comment:
nit pick: could we wrap this with `useCallback` so we avoid re-definiting them on each render?  Same for the `handleSecondaryCtaClick` one